sdk/cosmos/azure-cosmos-benchmark/src/main/java/com/azure/cosmos/benchmark/AsynReadWithMultipleClients.java

```diff
@@ -0,0 +1,295 @@
+        String csvFile = "clientHostAndKey.txt";
+        String line = "";
+        String splitBy = ";";
+        try (BufferedReader br = new BufferedReader(new FileReader(csvFile))) {
+            while ((line = br.readLine()) != null) {
+                String[] hostAndKey = line.split(splitBy);
+                if (hostAndKey.length >= 2) {
+                    String endpoint = hostAndKey[0].substring(hostAndKey[0].indexOf(ACCOUNT_ENDPOINT_TAG) + ACCOUNT_ENDPOINT_TAG.length());
+                    String key = hostAndKey[1].substring(hostAndKey[1].indexOf(ACCOUNT_KEY_TAG) + ACCOUNT_KEY_TAG.length());
+                    CosmosAsyncClient asyncClient = new CosmosClientBuilder()
+                        .endpoint(endpoint)
+                        .key(key)
+                        .connectionPolicy(configuration.getConnectionPolicy())
+                        .consistencyLevel(configuration.getConsistencyLevel())
+                        .connectionReuseAcrossClientsEnabled(true)
+                        .buildAsyncClient();
+                    List<PojoizedJson> docsToRead = new ArrayList<>();
+                    if (!configuration.isDeleteCollections()) {
+                        CosmosAsyncDatabase cosmosAsyncDatabase = asyncClient.createDatabaseIfNotExists(this.configuration.getDatabaseId()).block().getDatabase();
+                        CosmosAsyncContainer cosmosAsyncContainer = cosmosAsyncDatabase.createContainerIfNotExists(configuration.getCollectionId(), PARTITION_KEY, 100000).block().getContainer();
+                        String partitionKey = cosmosAsyncContainer.read().block().getProperties().getPartitionKeyDefinition()
+                            .getPaths().iterator().next().split("/")[1];
+                        String dataFieldValue = RandomStringUtils.randomAlphabetic(this.configuration.getDocumentDataFieldSize());
+                        ArrayList<Flux<PojoizedJson>> createDocumentObservables = new ArrayList<>();
+
+                        for (int i = 0; i < this.configuration.getNumberOfPreCreatedDocuments(); i++) {
+                            String uuid = UUID.randomUUID().toString();
+                            com.azure.cosmos.benchmark.PojoizedJson newDoc = generateDocument(uuid, dataFieldValue, partitionKey);
+
+                            Flux<PojoizedJson> obs = cosmosAsyncContainer.createItem(newDoc).map(resp -> {
+                                    com.azure.cosmos.benchmark.PojoizedJson x =
+                                        resp.getItem();
+                                    return x;
+                                }
+                            ).flux();
+                            createDocumentObservables.add(obs);
+                        }
+                        docsToRead = Flux.merge(Flux.fromIterable(createDocumentObservables), 100).collectList().block();
+
+                        logger.info("Client have been initialized with data created for host {}", hostAndKey[0]);
```

Comment:
this will print the key as well. can we just print the endpoint. ideally we should not log key anywhere in the log.

Reply:
discussed offline